compiler+runtime/src/cpp/jank/read/lex.cpp

```diff
@@ -440,81 +484,192 @@ namespace jank::read
           }
         /* Numbers. */
         case '-':
+          {
+            if(found_r)
+            {
+              ++pos;
+              return err(error{ token_start, pos, "invalid number: '-' after radix" });
+            }
+          }
         case '0' ... '9':
           {
-            auto &&e(check_whitespace(found_space));
-            if(e.is_some())
+            if(auto &&e(check_whitespace(found_space)); e.is_some())
             {
               return err(std::move(e.unwrap()));
             }
+
             native_bool contains_leading_digit{ file[token_start] != '-' };
             native_bool contains_dot{};
             native_bool is_scientific{};
             native_bool found_exponent_sign{};
             native_bool expecting_exponent{};
+            native_bool expecting_more_digits{};
+            int radix{ 10 };
+            auto r_pos{ pos }; /* records the 'r' position if one is found */
+            native_bool found_beginning_negative{};
+
+            if(auto const [c, l]{ peek().unwrap_or({ ' ', 1 }) };
+               file[token_start] == '-' && c == '0' && !found_slash_after_number)
+            {
+              contains_leading_digit = true;
+              if(auto const [f, l]{ peek(2).unwrap_or({ ' ', 1 }) }; f != ' ')
+              {
+                // auto const[n, _] {peek(3).unwrap_or({' ', 1})};
```

Comment:
Please remove this commented code, as well as the others in the various branches below.